pyasn1/codec/ber/decoder.py

```diff
@@ -1063,22 +1098,22 @@ def valueDecoder(self, substrate, asn1Spec,
             isUntagged = tagSet != asn1Spec.tagSet
 
         if isUntagged:
-            fullSubstrate = options['fullSubstrate']
+            fullPosition = options['fullPosition']
+            currentPosition = substrate.tell()
 
-            # untagged Any container, recover inner header substrate
-            length += len(fullSubstrate) - len(substrate)
-            substrate = fullSubstrate
+            substrate.seek(fullPosition, os.SEEK_SET)
```

Comment:
Oh, backtracking! Perhaps that's not always be possible to seek back, is it?

Reply:
You're right, probably we need to keep the extra bytes elsewhere (like the original "fullSubstrate" but smaller in memory).

Reply:
In the different implementation (https://github.com/etingof/pyasn1/pull/176), I envelop the non-seekable objects in BufferedReader.

Reply:
I am curious which streams are not seekable since we depend on that... From Python [docs](https://docs.python.org/3/library/io.html#io.IOBase.seekable) it seems some are not.